src/cloud_provider/aws/regions.rs

```diff
@@ -0,0 +1,373 @@
+    EuSouth1A,
+    EuSouth1B,
+    EuSouth1C,
+    // Bahrain
+    MeSouth1A,
+    MeSouth1B,
+    MeSouth1C,
+    // Sao Paulo
+    SaEast1A,
+    SaEast1B,
+    SaEast1C,
+}
+
+impl ToTerraformString for AwsZones {
+    fn to_terraform_format_string(&self) -> String {
+        format!("\"{}\"", self.to_string())
+    }
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Debug, Clone, EnumIter)]
+// Sync with Qovery Core team if you update this content
+pub enum AwsRegion {
+    UsEast1,
+    UsEast2,
+    UsWest2,
+    AfSouth1,
+    ApEast1,
+    ApSouth1,
+    ApNortheast1,
+    ApNortheast2,
+    ApNortheast3,
+    ApSoutheast1,
+    ApSoutheast2,
+    CaCentral1,
+    CnNorth1,
+    CnNorthwest1,
+    EuCentral1,
+    EuWest1,
+    EuWest2,
+    EuWest3,
+    EuNorth1,
+    EuSouth1,
+    MeSouth1,
+    SaEast1,
+}
+
+impl FromStr for AwsRegion {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<AwsRegion, ()> {
+        let v: &str = &s.to_lowercase();
+        match v {
+            "ap-east-1" | "apeast1" => Ok(AwsRegion::ApEast1),
+            "ap-northeast-1" | "apnortheast1" => Ok(AwsRegion::ApNortheast1),
+            "ap-northeast-2" | "apnortheast2" => Ok(AwsRegion::ApNortheast2),
+            "ap-northeast-3" | "apnortheast3" => Ok(AwsRegion::ApNortheast3),
+            "ap-south-1" | "apsouth1" => Ok(AwsRegion::ApSouth1),
+            "ap-southeast-1" | "apsoutheast1" => Ok(AwsRegion::ApSoutheast1),
+            "ap-southeast-2" | "apsoutheast2" => Ok(AwsRegion::ApSoutheast2),
+            "ca-central-1" | "cacentral1" => Ok(AwsRegion::CaCentral1),
+            "eu-central-1" | "eucentral1" => Ok(AwsRegion::EuCentral1),
+            "eu-west-1" | "euwest1" => Ok(AwsRegion::EuWest1),
+            "eu-west-2" | "euwest2" => Ok(AwsRegion::EuWest2),
+            "eu-west-3" | "euwest3" => Ok(AwsRegion::EuWest3),
+            "eu-north-1" | "eunorth1" => Ok(AwsRegion::EuNorth1),
+            "eu-south-1" | "eusouth1" => Ok(AwsRegion::EuSouth1),
+            "me-south-1" | "mesouth1" => Ok(AwsRegion::MeSouth1),
+            "sa-east-1" | "saeast1" => Ok(AwsRegion::SaEast1),
+            "us-east-1" | "useast1" => Ok(AwsRegion::UsEast1),
+            "us-east-2" | "useast2" => Ok(AwsRegion::UsEast2),
+            "us-west-2" | "uswest2" => Ok(AwsRegion::UsWest2),
+            "cn-north-1" | "cnnorth1" => Ok(AwsRegion::CnNorth1),
+            "cn-northwest-1" | "cnnorthwest1" => Ok(AwsRegion::CnNorthwest1),
+            "af-south-1" | "afsouth1" => Ok(AwsRegion::AfSouth1),
+            _ => Err(()),
+        }
+    }
+}
+
+impl AwsRegion {
+    pub fn new(&self) -> &AwsRegion {
+        self
+    }
+
+    pub fn to_string(&self) -> String {
+        let enum_name = format!("{}", self);
+        format!("{}", enum_name)
+    }
+
+    pub fn to_aws_format(&self) -> String {
+        let enum_name = format!("{}", self);
+        let re = Regex::new(r"^(.+)([A-Z])(.+)(\d)").expect("error while building the regex for AWS zones");
```

Comment:
Remove the regex  and create a custom mapper with a match on the enum.
Nobody will expect to break this function if it renames an enum.
also it will remove the need for the regex